interfaces/interface.js

```diff
@@ -30,12 +30,29 @@ export class AbstractCoreInterface {
   // system related
   // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
 
-  normalizeSync(name, parentName, isPlugin) { todo("normalizeSync") }
-  normalize(name, parent, parentAddress)    { todo("normalize") }
-  printSystemConfig()                       { todo("printSystemConfig") }
-  getConfig()                               { todo("getConfig") }
-  setConfig(conf)                           { todo("setConfig") }
-  getPackages()                             { todo("getPackages") }
+  normalizeSync(name, parentName, isPlugin) {
+    return modules.System.decanonicalize(name, parentName, isPlugin);
+  }
+  
+  normalize(name, parent, parentAddress) {
+    return modules.System.normalize(name, parent, parentAddress);
```

Comment:
I think I initially wrote these lines in a concrete implementation (hence the reference to `modules`). However, when I merged with master, these changes somehow ended up in the abstract class instead of the concrete subclasses. I guess I should hve paid more attention while merging.